File: Design-Patterns/Decorator/Cpp/character.cpp
#include "character.h"

character::character(std::shared_ptr<base_attributes> base_attributes)
{
	base_attributes_ = base_attributes;
	attributes_ = base_attributes;
}

int character::get_x()
{
	return x_;
}

int character::get_y()
{
	return y_;
}

std::shared_ptr<base_attributes> character::get_base_attributes()
{
	return base_attributes_;
}

std::shared_ptr<attributes> character::get_attributes()
{
	return attributes_;
}

void character::set_attributes(std::shared_ptr<attributes> attributes)
{
	attributes_ = attributes;
}

File: Design-Patterns/Decorator/CSharp/Character.cs
using System.Drawing;
using Decorator.Attributes;

namespace Decorator
{
    public class Character
    {
        public Point Position { get; protected set; }
        public BaseAttributes BaseAttributes { get; }
        public Attributes.Attributes Attributes { get; set; }

        public Character(BaseAttributes baseAttributes)
        {
            BaseAttributes = baseAttributes;
            Attributes = baseAttributes;
        }
    }
}
